src/Tomcat/TomcatController.ts

```diff
@@ -377,9 +377,12 @@ export class TomcatController {
         if (await fse.pathExists(path.join(folderLocation, 'META-INF', 'context.xml'))) {
             const xml: string = fs.readFileSync(path.join(folderLocation, 'META-INF', 'context.xml'), 'utf8');
             const jsonFromXml: any = await Utility.parseXml(xml);
-            if (jsonFromXml && jsonFromXml.Context && jsonFromXml.Context.$) {
-                const rawPath: string = jsonFromXml.Context.$.path;
-                appName = this.parseContextPathToFolderName(rawPath);
+            if (jsonFromXml) {
+                if (jsonFromXml.Context && jsonFromXml.Context.$) {
+                    appName = this.parseContextPathToFolderName(jsonFromXml.Context.$.path);
```

Comment:
If the `jsonFromXml.Context.$.path` is null or undefined, it will cause NPE in `parseContextPathToFolderName`.

Reply:
Ah shoot I originally was going to check for null in `parseContextPathToFolderName`. Change made.